Archive CI: only use token if needed

jaimergp opened the pull request:
Comes from https://github.com/conda-forge/admin-requests/pull/536#issuecomment-1371999295

Checking the status does not require a token, which is not present or needed on PRs.

jakirkham commented:
Thanks Jaime! 🙏

Happy to give this a try.

Do we run into any GitHub API limitations when just doing `GET`s without a token? Recall the rate limit was somewhat low when not using a token, but maybe that isn't an issue just for running the occasional test (unless we need to archive a bunch of feedstocks). Any other caveats?

jaimergp commented:
Thanks @jakirkham - I added a note about possible API rate limiting. I don't think we'll see enough volume here to notice but we'll see. Worst case we do export the public token in a future PR?

I also refactored a bit to ensure that all errors are caught (and not just the first one, which might be annoying for long queues if we ever have to mass archive).

jakirkham commented on archive_feedstock.py:
```diff
@@ -54,14 +56,26 @@ def feedstocks(directory="archive"):
+    for task in "archive", "unarchive":
+        seen = set()
+        for feedstock in feedstocks(task):
+            verb = "Archiving" if task == "archive" else "Unarchiving"
+            print(f"{verb} {owner}/{feedstock}-feedstock...", end=" ")
```
It looks like the verb is getting cut short. For [example]( https://github.com/conda-forge/admin-requests/actions/runs/3845806223/jobs/6550384893#step:6:12 ):

```python
Error: ng conda-forge/retworkx-feedstock... [ERROR]
```

jaimergp replied:
I think that's GH's rendering detecting `[ERROR]` and messing with the output?

jaimergp replied:
Let's see if I can use something else.

jakirkham replied:
It might also be that `print`ing is being buffered and so things are getting ordered in an odd way. If we disable buffering, this may help

jaimergp replied:
Added that too just in case!

jakirkham replied:
Interesting, looks like you were right. GHA is interpreting and modifying the line somehow. Using `!!` [works]( https://github.com/conda-forge/admin-requests/actions/runs/3845967004/jobs/6550724600#step:6:10 )

```python
Archiving conda-forge/doesnotexist-feedstock... [!!]
```

jakirkham replied:
Appears [this is intentional]( https://github.blog/2020-09-23-a-better-logs-experience-with-github-actions/#opening-the-door-to-a-more-colorful-experience ). Also looks like there is a way to [affect this behavior]( https://stackoverflow.com/a/71091060 )

jaimergp commented:
Yay we hit the rate limit 😂 Good news, we don't have to wait and see now.

jakirkham commented:
Oh it used to be more like 60 for unauthorized. Maybe that has gone down a lot more? Or we are using more requests than we think?

jaimergp commented:
The current workflow does work as intended for now, so up to you @jakirkham. We can always refine.

jakirkham commented:
Thanks Jaime 🙏

Yeah using a GH token seem like the right thing to do. The [default permissions]( https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token ) are only read for PRs from forks. Though we could tighten it down more if needed.

Am ok with what we have. Though will wait for @beckermr to comment

beckermr commented:
LGTM!

beckermr commented:
The standard GHA secrets are safe to use in PRs w/ forks.
